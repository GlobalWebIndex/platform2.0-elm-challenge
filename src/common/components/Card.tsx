import { Badge, Box } from '@chakra-ui/react';
import React from 'react';

interface Props {
    children: JSX.Element;
    badges?: string[];
}

export const Card = ({ children, badges = [] }: Props): JSX.Element => (
  <Box
    borderWidth="1px"
    borderRadius="lg"
    overflow="hidden"
    boxShadow="rgba(0, 0, 0, 0.1) 0px 4px 6px -1px, rgba(0, 0, 0, 0.06) 0px 2px 4px -1px;"
  >
    {children}
    {badges.length > 0 && (
    <Box p="3">
      <Box d="flex" alignItems="baseline">
        {
            badges.map((badgeText) => (
              <Badge key={badgeText} borderRadius="md" px="2" colorScheme="purple" variant="solid">
                {badgeText}
              </Badge>
            ))
        }
      </Box>
    </Box>
    )}
  </Box>
);
