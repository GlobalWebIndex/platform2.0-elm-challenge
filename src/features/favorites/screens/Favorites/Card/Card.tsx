import styled from 'styled-components';

import { IFavorite } from 'features/favorites/types';
import IconButton from 'common/components/IconButton';
import { Colors } from 'theme';

const Wrapper = styled.div`
  background-color: ${Colors.white};
  width: 180px;
  height: 240px;
  border-radius: 16px;
  overflow: hidden;
  flew-wrap: wrap;
  margin: 12px;
`;

const ImgWrapper = styled.div`
  width: 180px;
  height: 180px;
`;

const Img = styled.img`
  width: 180px;
  height: 180px;
  object-fit: cover;
`;

interface Props {
  favorite: IFavorite;
  onDelete: (id: string) => void;
}

const Card = ({ favorite, onDelete }: Props) => {
  const handleDelete = () => {
    onDelete(favorite.id);
  };
  return (
    <Wrapper className="flex flex-col items-center justify-between">
      <ImgWrapper>
        <Img src={favorite?.image?.url} alt="a cat" />
      </ImgWrapper>
      <div className="flex items-center justify-center bg-red-400 w-full h-24">
        <IconButton icon="delete" onClick={handleDelete} />
      </div>
    </Wrapper>
  );
};

export default Card;
