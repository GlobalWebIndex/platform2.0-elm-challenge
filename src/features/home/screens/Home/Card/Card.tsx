import styled from 'styled-components';

const Wrapper = styled.div`
  background-color: white;
  width: 200px;
  height: 200px;
`;

const SImg = styled.img`
  width: 180px;
  height: 180px;
  object-fit: contain;
  border: 2px solid red;
  border-radius: 16px;
`;

interface Props {
  url: string;
}

const Card = ({ url }: Props) => {
  return (
    <Wrapper>
      <SImg src={url} alt="a cat" />
    </Wrapper>
  );
};

export default Card;
