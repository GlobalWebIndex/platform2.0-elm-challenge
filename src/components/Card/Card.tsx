import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";

const CardStyle = {
  minWidth: 375,
  maxWidth: 325,
};

export default function BasicCard({ item }: { item: any }) {
  return (
    <Card sx={CardStyle} key={item.id}>
      <CardContent>
        <img alt="Cat" src={item.url} width="350" height="300" />
      </CardContent>
    </Card>
  );
}
