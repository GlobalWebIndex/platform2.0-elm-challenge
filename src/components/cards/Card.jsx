import React from "react";
import { Link, useLocation } from "react-router-dom";
import { useStyles } from "./Card.style";
import CardContents from "./CardContents";
import PropTypes from "prop-types";

const Card = ({ cardLink, isInGrid = false, ...restProps }) => {
  let classes = useStyles({ isInGrid });
  const location = useLocation();

  return cardLink ? (
    <Link
      className={classes.cardItem}
      to={cardLink}
      state={{ backgroundLocation: location }}
    >
      <CardContents {...restProps} />
    </Link>
  ) : (
    <div className={classes.cardItem}>
      <CardContents {...restProps} />
    </div>
  );
};

Card.propTypes = {
  imageSrc: PropTypes.string,
  imageHeight: PropTypes.number,
  imageWidth: PropTypes.number,
  id: PropTypes.string,
  title: PropTypes.string,
};

export default Card;
