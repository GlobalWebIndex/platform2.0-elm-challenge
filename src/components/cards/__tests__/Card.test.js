import { render } from "@testing-library/react";
import Card from "../Card";
import { BrowserRouter } from "react-router-dom";

const propsData = {
  imageSrc: "https://cat.com",
  imageWidth: 311,
  imageHeight: 111,
  id: "abc",
  title: "lovely cat",
};

describe("<Card/>", () => {
  it("renders a card with dedicated link", () => {
    const { container } = render(
      <Card cardLink="cat/abc" isInGrid {...propsData} />,
      { wrapper: BrowserRouter }
    );

    expect(container).toMatchSnapshot();
  });

  it("renders a card without dedicated link", () => {
    const { container } = render(<Card {...propsData} />, {
      wrapper: BrowserRouter,
    });

    expect(container).toMatchSnapshot();
  });
});
