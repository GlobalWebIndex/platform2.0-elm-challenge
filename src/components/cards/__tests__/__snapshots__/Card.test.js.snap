// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders a card with dedicated link 1`] = `
<div>
  <a
    class="cardItem-0-2-1 cardItem-d0-0-2-5"
    href="/cat/abc"
  >
    <img
      alt="id abc"
      class="cardImage-0-2-2"
      data-testid="card-image"
      height="111"
      loading="lazy"
      src="https://cat.com"
      width="311"
    />
    <div
      class="cardTitle-0-2-3"
      data-testid="card-title"
    >
      lovely cat
    </div>
  </a>
</div>
`;

exports[`<Card/> renders a card without dedicated link 1`] = `
<div>
  <div
    class="cardItem-0-2-1 cardItem-d2-0-2-7"
  >
    <img
      alt="id abc"
      class="cardImage-0-2-2"
      data-testid="card-image"
      height="111"
      loading="lazy"
      src="https://cat.com"
      width="311"
    />
    <div
      class="cardTitle-0-2-3"
      data-testid="card-title"
    >
      lovely cat
    </div>
  </div>
</div>
`;
